Restarted and Unsealed Vault v0.11.0 and paniced 'panic: interface conversion: interface {} is nil, not []interface {}'

**Describe the bug**
Restarted Vault (docker) v0.11.0 and unsealed vault. Right after I get a panic about interfaces being nil.

**To Reproduce**
Steps to reproduce the behavior:
1. Restart docker container
2. Unseal vault

**Expected behavior**
Vault is unsealed and available but instead it restarts and is sealed again.

**Environment:**
* Vault Server Version (retrieve with `vault status`): v0.11.0
* Vault CLI Version (retrieve with `vault version`): v0.11.0
* Server Operating System/Architecture: CoreOS Container Linux 1800.6.0

Vault server configuration file(s):

```json
VAULT_LOCAL_CONFIG=
      {
        "cluster_name": "vault-corp-prod",
        "ui":true,
        "cache_size": "32000",
        "listener": [
          {
            "tcp":
              {
                "address": "0.0.0.0:8200",
                "cluster_address": "0.0.0.0:8201",
                "tls_disable": "false",
                "tls_cert_file": "cert.crt",
                "tls_key_file": "cert.crt.key"
              }
          }
        ],
        "storage":
        {
          "consul":
            {
              "address": "consul-agent:8500",
              "service": "vault-corp-prod",
              "path": "vault/",
              "token": "",
              "api_addr": "https://:443",
              "cluster_addr": "https://:443",
              "disable_clustering": "false"
            }
        }
      }
```

**Additional context**

```
2018-09-04T16:00:49.584Z [INFO ] core: vault is unsealed
2018-09-04T16:00:49.585Z [INFO ] core: entering standby mode
2018-09-04T16:00:49.617Z [INFO ] core: acquired lock, enabling active operation
2018-09-04T16:00:49.723Z [INFO ] core: post-unseal setup starting
2018-09-04T16:00:49.728Z [INFO ] core: loaded wrapping token key
2018-09-04T16:00:49.728Z [INFO ] core: successfully setup plugin catalog: plugin-directory=
2018-09-04T16:00:49.742Z [INFO ] core: successfully mounted backend: type=kv path=secret/
2018-09-04T16:00:49.742Z [INFO ] core: successfully mounted backend: type=system path=sys/
2018-09-04T16:00:49.742Z [INFO ] core: successfully mounted backend: type=identity path=identity/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=pki path=pki/internal/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=pki path=pki/consul-corp-prod/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=pki path=pki/vault-corp-prod/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=consul path=consul/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=pki path=pki/elasticsearch-corp-prod/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=database path=database/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=ssh path=ssh/
2018-09-04T16:00:49.743Z [INFO ] core: successfully mounted backend: type=ssh path=ssh-client-signer/
2018-09-04T16:00:49.744Z [INFO ] core: successfully mounted backend: type=pki path=aaron_internal/
2018-09-04T16:00:49.744Z [INFO ] core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2018-09-04T16:00:49.779Z [INFO ] core: restoring leases
2018-09-04T16:00:49.780Z [INFO ] rollback: starting rollback manager
2018-09-04T16:00:49.817Z [INFO ] identity: entities restored
2018-09-04T16:00:49.841Z [INFO ] identity: groups restored
2018-09-04T16:00:49.845Z [INFO ] core: post-unseal setup complete
2018-09-04T16:00:49.845Z [INFO ] core: core/startClusterListener: starting listener: listener_address=0.0.0.0:8201
2018-09-04T16:00:49.845Z [INFO ] core: core/startClusterListener: serving cluster requests: cluster_listen_address=[::]:8201
2018-09-04T16:00:49.937Z [INFO ] expiration: lease restore complete
panic: interface conversion: interface {} is nil, not []interface {}

goroutine 488 [running]:
github.com/hashicorp/vault/builtin/logical/database.(*databaseBackend).secretCredsRevoke.func1(0x28fa8a0, 0xc420584600, 0xc42001bd40, 0xc42058bf20, 0x0, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/builtin/logical/database/secret_creds.go:114 +0x916
github.com/hashicorp/vault/logical/framework.(*Secret).HandleRevoke(0xc4208b0cf0, 0x28fa8a0, 0xc420584600, 0xc42001bd40, 0xc42001bcd8, 0x1000b0c69d68301, 0x0)
	/gopath/src/github.com/hashicorp/vault/logical/framework/secret.go:87 +0x9a
github.com/hashicorp/vault/logical/framework.(*Backend).handleRevokeRenew(0xc4202d3ba0, 0x28fa8a0, 0xc420584600, 0xc42001bd40, 0x323039610422c72a, 0x6530342d61303366, 0xc420b91898)
	/gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:395 +0x1fb
github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc4202d3ba0, 0x28fa8a0, 0xc420584600, 0xc42001bd40, 0x0, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:170 +0x6fb
github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc42057ea00, 0x28fa8a0, 0xc420584600, 0xc42001bd40, 0x0, 0x0, 0x0, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/vault/router.go:523 +0x7c2
github.com/hashicorp/vault/vault.(*Router).Route(0xc42057ea00, 0x28fa8a0, 0xc420584600, 0xc42001bd40, 0x1c8a577, 0xc4205903c8, 0xc4207f0dc0)
	/gopath/src/github.com/hashicorp/vault/vault/router.go:378 +0x4e
github.com/hashicorp/vault/vault.(*ExpirationManager).revokeEntry(0xc420590360, 0x28fa920, 0xc4207c1440, 0xc4207138c0, 0x42, 0xc4207138c0)
	/gopath/src/github.com/hashicorp/vault/vault/expiration.go:1159 +0x1ca
github.com/hashicorp/vault/vault.(*ExpirationManager).revokeCommon(0xc420590360, 0x28fa920, 0xc4207c1440, 0xc4207f0dc0, 0x42, 0xbedbc8ec752f0000, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/vault/expiration.go:548 +0x4ff
github.com/hashicorp/vault/vault.(*ExpirationManager).Revoke(0xc420590360, 0x28fa920, 0xc4207c1440, 0xc4207f0dc0, 0x42, 0x0, 0x0)
	/gopath/src/github.com/hashicorp/vault/vault/expiration.go:495 +0x12c
github.com/hashicorp/vault/vault.(*ExpirationManager).expireID(0xc420590360, 0xc4207f0dc0, 0x42)
	/gopath/src/github.com/hashicorp/vault/vault/expiration.go:1133 +0x31c
github.com/hashicorp/vault/vault.(*ExpirationManager).updatePendingInternal.func1()
	/gopath/src/github.com/hashicorp/vault/vault/expiration.go:1091 +0x3f
created by time.goFunc
	/goroot/src/time/sleep.go:172 +0x44
```

I downgraded to v0.10.2 and it was able to start. When it started up the database engine tried to revoke stuff but did have errors.

```
2018-09-04T17:10:00.265Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/db/24d1bbfb-61dd-c0c6-369a-5f1531f9b5c5 error="failed to revoke entry: resp: (*logical.Response)(nil) err: error during revoke: could not find role with name "db""
2018-09-04T17:10:04.542Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/db/aba22b7d-b8c5-3881-e4e7-486362c94445 error="failed to revoke entry: resp: (*logical.Response)(nil) err: sql: Scan error on column index 1: unsupported Scan, storing driver.Value type <nil> into type *string"
2018-09-04T17:10:04.547Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/db/3fcb3083-d4b2-19bf-018e-84d7343d3d4b error="failed to revoke entry: resp: (*logical.Response)(nil) err: sql: Scan error on column index 1: unsupported Scan, storing driver.Value type <nil> into type *string"
2018-09-04T17:10:10.275Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/db/24d1bbfb-61dd-c0c6-369a-5f1531f9b5c5 error="failed to revoke entry: resp: (*logical.Response)(nil) err: error during revoke: could not find role with name "db""
2018-09-04T17:10:16.022Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/db/3fcb3083-d4b2-19bf-018e-84d7343d3d4b error="failed to revoke entry: resp: (*logical.Response)(nil) err: sql: Scan error on column index 1: unsupported Scan, storing driver.Value type <nil> into type *string"
2018-09-04T17:10:16.091Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/db/aba22b7d-b8c5-3881-e4e7-486362c94445 error="failed to revoke entry: resp: (*logical.Response)(nil) err: sql: Scan error on column index 1: unsupported Scan, storing driver.Value type <nil> into type *string"
```

--- a/builtin/logical/database/backend_test.go
+++ b/builtin/logical/database/backend_test.go
@@ -604,7 +604,7 @@ func TestBackend_basic(t *testing.T) {
 					"username":              credsResp.Data["username"],
 					"role":                  "plugin-role-test",
 					"db_name":               "plugin-test",
-					"revocation_statements": []interface{}(nil),
+					"revocation_statements": nil,
 				},
 			},
 		})

--- a/builtin/logical/database/secret_creds.go
+++ b/builtin/logical/database/secret_creds.go
@@ -111,8 +111,15 @@ func (b *databaseBackend) secretCredsRevoke() framework.OperationFunc {
 			if statementsRaw, ok := req.Secret.InternalData["revocation_statements"]; !ok {
 				return nil, fmt.Errorf("error during revoke: could not find role with name %q or embedded revocation statement data", req.Secret.InternalData["role"])
 			} else {
-				for _, v := range statementsRaw.([]interface{}) {
-					statements.Revocation = append(statements.Revocation, v.(string))
+				if statementsRaw != nil {
+					statementsSlice, ok := statementsRaw.([]interface{})
+					if !ok {
+						return nil, fmt.Errorf("error during revoke: could not find role with name %q and embedded reovcation data could not be read", req.Secret.InternalData["role"])
+					} else {
+						for _, v := range statementsSlice {
+							statements.Revocation = append(statements.Revocation, v.(string))
+						}
+					}
 				}
 			}
 		}

--- a/builtin/logical/database/secret_creds.go
+++ b/builtin/logical/database/secret_creds.go
@@ -111,6 +111,9 @@ func (b *databaseBackend) secretCredsRevoke() framework.OperationFunc {
 			if statementsRaw, ok := req.Secret.InternalData["revocation_statements"]; !ok {
 				return nil, fmt.Errorf("error during revoke: could not find role with name %q or embedded revocation statement data", req.Secret.InternalData["role"])
 			} else {
+				// If we don't actually have any statements, because none were
+				// set in the role, we'll end up with an empty one and the
+				// default for the db type will be attempted
 				if statementsRaw != nil {
 					statementsSlice, ok := statementsRaw.([]interface{})
 					if !ok {

